testsuite/MDAnalysisTests/core/test_universe.py

```diff
@@ -460,11 +474,17 @@ def _check_atomgroup(self, ag, u):
 
     def test_atomgroup_guess_bonds(self):
         """Test an atomgroup doing guess bonds"""
-        u = mda.Universe(two_water_gro)
-
-        ag = u.atoms[:3]
-        ag.guess_bonds()
-        self._check_atomgroup(ag, u)
+        def multi_parameters_atomgroup(ff=0.55, lb=0.1, nbonds=2):
+            u = mda.Universe(two_water_gro)
+            ag = u.atoms[:3]
+            ag.guess_bonds(fudge_factor=ff, lower_bound=lb)
+            assert_equal(len(ag.bonds), nbonds)
+
+        # We run here multiple tests with different values for
+        # fudge_factor and lower_bound
+        multi_parameters_atomgroup(0.55, 0.1, 2)
+        multi_parameters_atomgroup(0.9, 1.6, 1)
+        multi_parameters_atomgroup(0.5, 0.2, 2)
```

Comment:
You should be able to do what `multi_parameters_atomgroup` does with [pytest fixtures](https://docs.pytest.org/en/6.2.x/parametrize.html). It will create separate tests that can fail individually.